app/controllers/schools_controller.rb

```diff
@@ -12,11 +12,11 @@ def show
   end
 
   def search
-    School.all.collect { |school| ["#{school.name}, #{school.city}, #{school.state}", school.name] }
+    School.all.collect { |school| ["#{school.name}, #{school.country}, #{school.city}, #{school.state}", school.name] }
```

Comment:
Hmm. OK, I think we might want a different method here. 

The presentation should definitely be:
"School Name, City, State, Country" if we have all the data.
I wonder if the format is actually best as:
"School Name (City, State)" if the country is USA
and 
"School Name (City, State, Country)" when the country is not the US.

That logic is a little complex, so it might make sense to write a different method...

```
     def format_school([name, city, state, country]):
        country_str = country == "US" ? '' : ", #{country}"
        "#{name} (#{city}, #{state}#{country_str})"
    end

    # This should be a model method, perhaps called self.search_list on the School model.
    School.all.pluck(:name, :city, :state, :country).map { |data| [ format_state(data), name ] }
```
Use `pluck` here isn't strictly necessary, but it prevents Rails from creating 1000's of additional objects which uses a lot less memory.


```suggestion
    School.search_list
```